ERCS/erc-7802.md

```diff
@@ -0,0 +1,270 @@
+The local mint and burn functions remain unchanged. 
+Using dedicated functions for cross-chain operations provides a more modular approach, avoiding the need to modify the base implementation for each chain.
+
+**Dedicated events**
+
+A similar reasoning applies to having dedicated cross-chain-specific events. The separation significantly facilitates the work of indexers, analytics tools, and auditors. It allows for straightforward tracking of cross-chain activities, detecting anomalies, and monitoring bridge operations. If cross-chain and local events are indistinguishable, off-chain agents must implement complex logic to differentiate them, increasing the potential for errors and inefficiencies.
+
+### ERC-165 Interface
+
+The inclusion of ERC-165 provides an additional security check for integrators. By providing the interface identifier through the `supportsInterface` method, callers can programmatically confirm that the token adheres to the `IERC7802` interface. 
+This verification ensures that the token supports both `crosschainMint` and `crosschainBurn` functions, preventing scenarios where only one function is implemented. Such incomplete implementations could lead to issues like users burning tokens to bridge out but being unable to mint them upon return, resulting in failed cross-chain actions.
+
+It is important to note that this check can only be performed locally on the chain where the token contract resides. There is no inherent guarantee that the token on the receiving chain also supports the `IERC7802` interface. Ensuring cross-chain consistency of interface support is the responsibility of the bridge implementation.
+
+## Backwards Compatibility
+
+This proposal is fully backwards compatible with [ERC-20](./eip-20.md).
+
+As discussed in the Motivation section, a minimal, flexible cross-chain standard interface is necessary. The problem becomes larger as more tokens are deployed without a standardized format.
+
+- Upgradable tokens can be upgraded to implement the new interface.
+- Non-upgradable tokens cannot implement the interface on the token itself. They can still migrate to a standard-compliant version using a lockbox mechanism, as proposed by xERC-20. The idea is to lock non-mintable tokens and mint the same amount of interface-compliant tokens. The bridge contract can act as a lockbox on the native chain.
+
+Bridge contracts will also need an upgrade to integrate with the interface. Most popular bridges in terms of liquidity at the date are upgradable or use tokens that can be upgraded to point to new bridge contracts. 
+
+1. Arbitrum One ($5.23B)
+2. Portal Wormhole ($2.43B)
+3. Polygon PoS ($2.15B)
+4. OP Mainnet ($2.09B)
+5. Mantle ($983M)
+6. LayerZero OFTs ($465M)
+7. Omnibridge ($449M)
+8. Base ($350M)
+9. Scroll ($128M)
+10. zkSync ($77M)
+
+Source: L2Beat (10/31/24)
+
+## Reference Implementation
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code inline in the document must be CC0-1.0 licensed. If you have to keep it MIT, we do permit that in the assets directory.

Reply:
fixed